include/tsid/tasks/task-contact-force-equality.hpp

```diff
@@ -15,15 +15,87 @@
 // <http://www.gnu.org/licenses/>.
 //
 
-#ifndef __invdyn_tasks_fwd_hpp__
-#define __invdyn_tasks_fwd_hpp__
+#ifndef __invdyn_task_contact_force_equality_hpp__
+#define __invdyn_task_contact_force_equality_hpp__
+
+#include "tsid/math/fwd.hpp"
+#include "tsid/tasks/task-contact-force.hpp"
+#include "tsid/trajectories/trajectory-base.hpp"
+#include "tsid/math/constraint-equality.hpp"
+#include "tsid/formulations/inverse-dynamics-formulation-base.hpp"
 
 namespace tsid
 {
   namespace tasks
   {
-    
+
+    class TaskContactForceEquality : public TaskContactForce
+    {
+    public:
+      EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+      typedef math::Index Index;
+      typedef trajectories::TrajectorySample TrajectorySample;
+      typedef math::Vector Vector;
+      typedef math::Vector6 Vector6;
+      typedef math::Vector3 Vector3;
+      typedef math::ConstraintEquality ConstraintEquality;
+      typedef pinocchio::SE3 SE3;
+
+      TaskContactForceEquality(const std::string & name,
+                      		     RobotWrapper & robot,
+                               const double dt,
+                      		     const std::string & contactName="");
+
+      void setContactList(const std::vector<std::shared_ptr<ContactLevel> >  *contacts);
```

Comment:
I think the TaskContactForceEquality shouldn't need to know the whole list of contacts. It should just provide a specification of the task related to its own contact force, for which it just needs to know the size of its own contact force vector. Then it's the Formulation class that should take care of putting this smaller matrix in the right spot of the larger matrix that multiplies the whole force vector.

Reply:
Yes I agree.
I think I did this because in the `inverse-dynamics-formulation-acc-force.cpp` the method used to compute the constraint for the force tasks takes the whole contacts list in argument for the CoP task:
https://github.com/stack-of-tasks/tsid/blob/096932896c4ef77f0f04a13bd8c4232b4a562151/src/formulations/inverse-dynamics-formulation-acc-force.cpp#L397

I change my method so that it only checks if the associated contact of the `TaskContactForceEquality` is in the list of contacts of the formulation problem. If it is, it uses the classical compute method; if not, it returns an error message and the empty constraint.

Reply:
Ok, sounds good to me!